src/sentry/api/endpoints/external_team.py

```diff
@@ -0,0 +1,82 @@
+import logging
+
+from django.db import IntegrityError
+
+from rest_framework import serializers, status
+from rest_framework.response import Response
+from sentry.api.serializers.rest_framework.base import CamelSnakeModelSerializer
+
+from sentry.api.bases.team import TeamEndpoint
+from sentry.api.serializers import serialize
+from sentry.models import ExternalTeam, EXTERNAL_PROVIDERS
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalTeamSerializer(CamelSnakeModelSerializer):
+    team_id = serializers.IntegerField(required=True)
+    external_id = serializers.CharField(required=True)
+    provider = serializers.ChoiceField(choices=list(EXTERNAL_PROVIDERS.values()))
+
+    class Meta:
+        model = ExternalTeam
+        fields = ["team_id", "external_id", "provider"]
+
+    def validate_provider(self, provider):
+        if provider not in EXTERNAL_PROVIDERS.values():
+            raise serializers.ValidationError(
+                f'The provider "{provider}" is not supported. We currently accept Github and Gitlab team identities.'
+            )
+        inv_providers_map = {v: k for k, v in EXTERNAL_PROVIDERS.items()}
+        return inv_providers_map[provider].value
+
+    def create(self, validated_data):
+        try:
+            return ExternalTeam.objects.create(**validated_data)
+        except IntegrityError:
+            # swallow the error and find and return the object.
+            # we currently expect this endpoint to be hit by start-up scripts, so if the association already exists, we'll just return it.
+            return list(ExternalTeam.objects.filter(**validated_data))
+
+    def update(self, instance, validated_data):
+        if "id" in validated_data:
+            validated_data.pop("id")
+        for key, value in validated_data.items():
+            setattr(self.instance, key, value)
+        try:
+            self.instance.save()
+            return self.instance
+        except IntegrityError:
+            raise serializers.ValidationError(
+                "There already exists an external team association with this external_id and provider."
+            )
+
+
+class ExternalTeamEndpoint(TeamEndpoint):
+    def post(self, request, team):
+        """
+        Create an External Team
+        `````````````
+
+        Update various attributes and configurable settings for the given
```

Comment:
This looks like a copy/paste from the PUT endpoint, shouldn't it mention creating?

Reply:
ahh good catch!! copy paste pretends to be my friend